File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export function decodedValue(colors: string[]) {
  const [color1, color2] = colors;
  const code1 = COLORS.indexOf(color1);
  const code2 = COLORS.indexOf(color2);
  return +`${code1}${code2}`;
};

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];

File: solutions/elixir/resistor-color-duo/1/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @colors %{
    :black => 0,
    :brown => 1,
    :red => 2,
    :orange => 3,
    :yellow => 4,
    :green => 5,
    :blue => 6,
    :violet => 7,
    :grey => 8,
    :white => 9
  }
  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value(colors) do
    [color1, color2 | _] = colors
    code1 = Map.get(@colors, color1)
    code2 = Map.get(@colors, color2)
    {res, _} = Integer.parse("#{code1}#{code2}")
    res
  end
end
